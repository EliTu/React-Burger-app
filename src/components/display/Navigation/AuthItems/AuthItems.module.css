@import '../../../../variables.css';

.AuthItems {
    display: block;
    margin: 0 1rem;
    padding: 0;
    list-style: none;
    align-items: center;
    text-align: center;
    height: 100%;
    width: 90%;
}

.LoggedInMessage {
    color: var(--coloring-bright-light);
    display: inline-block;
    height: 100%;
    font-size: 1rem;
    padding: 0.9rem 1rem;
    margin: auto 0;
}

.LoggedInMessage span {
    color: var(--glow-green-darker);
    text-align: center;
}

/* Medium screen */
@media only (min-width: 500px) and (max-width: 850px) {
    .AuthItems .LoggedInMessage {
        width: 60%;
        font-size: 0.9rem;
        padding: 0.9rem 0;
    }
}

/* Desktop */
@media (min-width: 500px) {
    .AuthItems {
        display: flex;
        flex-flow: row;
        justify-items: center;
        width: 100%;
    }
}