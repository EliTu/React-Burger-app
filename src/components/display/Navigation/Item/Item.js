import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { signInToggleClick } from '../AuthItems/store/actions';
import { authSignout } from '../../../containers/Authentication/store/actions';
import Icon from '../../../UI/Icon/Icon';
import { faSignInAlt } from '@fortawesome/free-solid-svg-icons';
import styles from './Item.module.css';
import PropTypes from 'prop-types';

const Item = ({
	link,
	children,
	isSignInDisplayed,
	signInItem,
	toggleSignIn,
	signOutItem,
	signOutClicked,
	isLoggedIn,
}) => {
	// CSS Modules styles:
	const { Item, active, AuthSignin, SigninActive } = styles;

	// Handle toggling the display property of the LogIn component
	const handleSignInToggleClick = event => {
		event.preventDefault();
		if (!isSignInDisplayed) toggleSignIn();
	};

	// Handle loging user out upon click on 'Log out':
	const handleLogOutClick = event => {
		event.preventDefault();
		signOutClicked('signOut');
	};

	// Set Sign in tab custom styles:
	const signInDefaultStyle = !isLoggedIn ? AuthSignin : null;
	const signInActiveStyle = isSignInDisplayed ? SigninActive : null;

	const authType =
		signInItem || signOutItem ? (
			<a
				className={[Item, signInDefaultStyle, signInActiveStyle].join(
					' '
				)}
				href={link}
				onClick={event =>
					signInItem
						? handleSignInToggleClick(event)
						: handleLogOutClick(event)
				}
			>
				{signInItem && <Icon iconType={faSignInAlt} />}
				{signInItem ? 'Sign in' : 'Log out'}
			</a>
		) : (
			<NavLink className={Item} activeClassName={active} to={link} exact>
				{children}
			</NavLink>
		);

	return <div className={Item}>{authType}</div>;
};

Item.propTypes = {
	link: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
	signInItem: PropTypes.bool,
	isSignInDisplayed: PropTypes.bool,
	toggleSignIn: PropTypes.func,
	signOutClicked: PropTypes.func,
	isLoggedIn: PropTypes.bool,
};

// Redux setup:
const mapStateToProps = state => {
	return {
		isSignInDisplayed: state.signIn.isSignInDisplayed,
		isLoggedIn: state.auth.isLoggedIn,
	};
};
const mapDispatchToProps = dispatch => {
	return {
		toggleSignIn: () => dispatch(signInToggleClick()),
		signOutClicked: signOut => dispatch(authSignout(signOut)),
	};
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Item);
