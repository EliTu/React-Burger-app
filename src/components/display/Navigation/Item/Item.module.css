@import '../../../../variables.css';

.Item {
	box-sizing: border-box;
	font-size: 1.3rem;
	margin: 0.8rem auto;
	width: 75%;
	display: block;
	transition: 0.5s all;
}

.Item a {
	width: 100%;
	align-self: auto;
	color: var(--coloring-light);
	text-decoration: none;
	text-align: center;
	display: block;
}

.Item .AuthSignin {
	color: var(--glow-green-darker);
}

.Item a:hover {
	border-bottom: 5px solid var(--coloring-bright-light);
	background-color: rgba(219, 221, 219, 0.144);
}

.Item a:active,
.Item a.active {
	background-color: rgba(0, 0, 0, 0);
	border-bottom: 10px solid var(--coloring-bright-light);
	color: var(--coloring-bright-light);
	background-color: rgba(219, 221, 219, 0.144);
}

/* Large screen */
@media screen and (min-width: 500px) {
	.Item {
		font-size: initial;
		margin: 0;
		height: 100%;
		width: auto;
		display: flex;
		align-items: center;
		transition: 0.3s all;
	}

	.Item a {
		align-self: flex-end;
		font-size: 1rem;
		align-self: auto;
		color: var(--coloring-light);
		height: 100%;
		width: auto;
		text-align: center;
		padding: 0.9rem 1rem;
		border-bottom: 4px solid transparent;
		border-radius: 0;
	}

	.Item .SigninActive {
		color: rgb(99, 248, 99);
		border-bottom: 10px solid var(--coloring-bright-light);
	}
}

/*  Medium screne */
@media only screen and (max-width: 850px) {
	.Item a {
		font-size: 0.9rem;
		width: 100%;
		padding: 0.9rem 0.4rem;
	}
}

/* small screen */
@media screen and (max-width: 550px) {
	.Item a {
		font-size: 1rem;
		width: 100%;
		padding: 0.9rem 0.4rem;
	}

	.Item a:active,
	.Item a.active {
		background-color: rgba(0, 0, 0, 0);
		border-bottom: 3px solid var(--glow-green-darker);
		color: var(--coloring-bright-light);
	}
}