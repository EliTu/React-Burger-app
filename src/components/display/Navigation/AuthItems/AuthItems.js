import React from 'react';
import { connect } from 'react-redux';
import Item from '../Item/Item';
import styles from './AuthItems.module.css';
import PropTypes from 'prop-types';
import Icon from '../../../UI/Icon/Icon';
import { faUserPlus } from '@fortawesome/free-solid-svg-icons';

export const AuthItems = ({ isLoggedIn, email }) => {
	// CSS Modules style:
	const { AuthItems, LoggedInMessage } = styles;

	return (
		<div className={AuthItems}>
			{!isLoggedIn ? (
				<>
					<Item signInItem link="">
						Sign in
					</Item>
					<Item link="/signup">
						{' '}
						<Icon iconType={faUserPlus} />
						Sign up
					</Item>
				</>
			) : (
				<>
					<p className={LoggedInMessage}>
						Welcome, <span>{email}</span>!
					</p>
					<Item signOutItem link="">
						Log out
					</Item>
				</>
			)}
		</div>
	);
};

AuthItems.propTypes = {
	isLoggedIn: PropTypes.bool,
	email: PropTypes.string,
};

// Redux setup:
const mapStateToProps = state => {
	return {
		isLoggedIn: state.auth.isLoggedIn,
		email: state.auth.email,
	};
};

export default connect(mapStateToProps)(AuthItems);
