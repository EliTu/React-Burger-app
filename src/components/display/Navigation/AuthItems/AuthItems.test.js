import React from 'react';
import { shallow } from 'enzyme';
import { AuthItems } from './AuthItems';
import Item from '../Item/Item';

describe('AuthItems', () => {
	let component;
	beforeEach(() => {
		component = shallow(<AuthItems />);
	});

	it('Should render 2 <Item> if the user is not signed in', () => {
		component.setProps({ isLoggedIn: false });

		expect(component).toMatchSnapshot();
		expect(component.find(Item)).toHaveLength(2);
		expect(component.find(Item)).not.toHaveLength(3);
		expect(component.find(Item)).not.toBeFalsy();
	});

	it('should render 1 <Item></Item> and if user is signed in ', () => {
		component.setProps({ isLoggedIn: true });

		expect(component).toMatchSnapshot();
		expect(component.find(Item)).toHaveLength(1);
		expect(component.find(Item)).not.toHaveLength(2);
		expect(component.find(Item)).not.toBeFalsy();
	});

	it('should contain an <Item>Log out</Item> if the user is authenticated and logged in', () => {
		component.setProps({ isLoggedIn: true });

		expect(component).toMatchSnapshot();
		expect(component.find(Item)).toBeTruthy();
		expect(
			component.contains(
				<Item signOutItem link="">
					Log out
				</Item>
			)
		).toEqual(true);
	});
});
