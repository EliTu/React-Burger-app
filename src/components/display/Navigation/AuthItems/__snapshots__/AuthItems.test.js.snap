// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthItems Should render 2 <Item> if the user is not signed in 1`] = `
<div
  className="AuthItems"
>
  <Connect(Item)
    link=""
    signInItem={true}
  >
    Sign in
  </Connect(Item)>
  <Connect(Item)
    link="/signup"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f234",
            "M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
          ],
          "iconName": "user-plus",
          "prefix": "fas",
        }
      }
    />
    Sign up
  </Connect(Item)>
</div>
`;

exports[`AuthItems should contain an <Item>Log out</Item> if the user is authenticated and logged in 1`] = `
<div
  className="AuthItems"
>
  <p
    className="LoggedInMessage"
  >
    Welcome, 
    <span />
    !
  </p>
  <Connect(Item)
    link=""
    signOutItem={true}
  >
    Log out
  </Connect(Item)>
</div>
`;

exports[`AuthItems should render 1 <Item></Item> and if user is signed in  1`] = `
<div
  className="AuthItems"
>
  <p
    className="LoggedInMessage"
  >
    Welcome, 
    <span />
    !
  </p>
  <Connect(Item)
    link=""
    signOutItem={true}
  >
    Log out
  </Connect(Item)>
</div>
`;
